include/openthread/tcat.h

```diff
@@ -0,0 +1,173 @@
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief
+ *  This file defines the top-level functions for the OpenThread TCAT.
+ *
+ *  @note
+ *   The functions in this module require the build-time feature `OPENTHREAD_CONFIG_BLE_TCAT_ENABLE=1`.
+ *
+ *  @note
+ *   To enable cipher suite DTLS_PSK_WITH_AES_128_CCM_8, MBEDTLS_KEY_EXCHANGE_PSK_ENABLED
+ *    must be enabled in mbedtls-config.h
+ *   To enable cipher suite DTLS_ECDHE_ECDSA_WITH_AES_128_CCM_8,
+ *    MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED must be enabled in mbedtls-config.h.
+ */
+
+#ifndef OPENTHREAD_TCAT_H_
+#define OPENTHREAD_TCAT_H_
+
+#include <stdint.h>
+#include <openthread/message.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @addtogroup api-ble-secure
+ *
+ * @brief
+ *   This module includes functions that implement TCAT communication.
+ *
+ *   The functions in this module are available when TCAT feature
+ *   (`OPENTHREAD_CONFIG_BLE_TCAT_ENABLE`) is enabled.
+ *
+ * @{
+ *
+ */
+
+#define OT_TCAT_MAX_SERVICE_NAME_LENGTH \
+    15 ///< Maximum string length of a UDP or TCP service name (does not include null char).
+
+/**
+ * Represents TCAT status code.
+ *
+ */
+typedef enum otTcatStatusCode
+{
+    OT_TCAT_STATUS_SUCCESS       = 0, ///< Command or request was successfully processed
+    OT_TCAT_STATUS_UNSUPPORTED   = 1, ///< Requested command or received TLV is not supported
+    OT_TCAT_STATUS_PARSE_ERROR   = 2, ///< Request / command could not be parsed correctly
+    OT_TCAT_STATUS_VALUE_ERROR   = 3, ///< The value of the transmitted TLV has an error
+    OT_TCAT_STATUS_GENERAL_ERROR = 4, ///< An error not matching any other category occurred
+    OT_TCAT_STATUS_BUSY          = 5, ///< Command cannot be executed because the resource is busy
+    OT_TCAT_STATUS_UNDEFINED    = 6, ///< The requested value, data or service is not defined (currently) or not present
+    OT_TCAT_STATUS_HASH_ERROR   = 7, ///< The hash value presented by the commissioner was incorrect
+    OT_TCAT_STATUS_UNAUTHORIZED = 8, ///< Sender does not have sufficient authorization for the given command
+
+} otTcatStatusCode;
+
+/**
+ * Represents TCAT status.
+ *
+ */
+typedef enum otTcatMessageType
+{
+    OT_TCAT_MESSAGE_TYPE_NONE   = 0, ///< Message which has been sent without activating the TCAT agent
+    OT_TCAT_MESSAGE_TYPE_STATUS = 1, ///< Message containing a status code (byte) as defined in otTcatStatusCode
+    OT_TCAT_MESSAGE_TYPE_UDP    = 2, ///< Message directed to a UDP service
+    OT_TCAT_MESSAGE_TYPE_TCP    = 3, ///< Message directed to a TCP service
+    OT_TCAT_MESSAGE_TYPE_CHANGED_TO_UDP_SERVICE = 4, ///< Client has changed to a UDP service
+    OT_TCAT_MESSAGE_TYPE_CHANGED_TO_TCP_SERVICE = 5, ///< Client has changed to a TCP service
+
+} otTcatMessageType;
+
+/**
+ * Represents TCAT a command class.
+ *
+ */
+typedef enum otTcatCommandClass
+{
+    OT_TCAT_COMMAND_CLASS_GENERAL         = 1, ///< TCAT commands related to general operations
```

Comment:
I'm updating the values here based on the latest draft4 spec! By copying the 0bxxx binary values directly from the spec, there should be a lower chance of confusion and error.

```suggestion
    OT_TCAT_COMMAND_CLASS_GENERAL         = 0b000, ///< TCAT commands related to general operations
```

Reply:
fixed.